colorcoder/MajorColor.java

```diff
@@ -0,0 +1,24 @@
+package colorcoder;
+
+enum MajorColor {
+        WHITE(0),
+        RED(1),
+        BLACK(2),
+        YELLOW(3),
+        VIOLET(4);
+        private int index;
+        private MajorColor(int index) {
+            this.index = index;
+        }
+        int getIndex() {
+            return index;
+        }
+        public static MajorColor fromIndex(int index) {
```

Comment:
This is very similar to the fromIndex in MinorColor. Try making a common function and re-use